src/components/Attachment/components/WaveProgressBar.tsx

```diff
@@ -0,0 +1,191 @@
+  waveformData: number[];
+  /** Allows to specify the number of bars into which the original waveformData array should be resampled */
+  amplitudesCount?: number;
+  /** Progress expressed in fractional number value btw 0 and 100. */
+  progress?: number;
+};
+export const WaveProgressBar = ({
+  amplitudesCount = 40,
+  progress = 0,
+  seek,
+  waveformData,
+}: WaveProgressBarProps) => {
+  const [progressIndicator, setProgressIndicator] = useState<HTMLDivElement | null>(null);
+  const isDragging = useRef(false);
+
+  const handleMouseDown = () => {
+    if (!progressIndicator) return;
+    isDragging.current = true;
+    progressIndicator.style.cursor = 'grabbing';
+  };
+
+  const handleMouseMove: MouseEventHandler<HTMLDivElement> = (event) => {
+    if (!isDragging.current) return;
+    seek(event);
+  };
+
+  const handleMouseUp = () => {
+    if (!progressIndicator) return;
+    isDragging.current = false;
+    progressIndicator.style.removeProperty('cursor');
+  };
+
+  const resampledWaveformData = useMemo(
+    () =>
+      waveformData.length === amplitudesCount
+        ? waveformData
+        : waveformData.length > amplitudesCount
+        ? downSample(waveformData, amplitudesCount)
+        : upSample(waveformData, amplitudesCount),
+    [amplitudesCount, waveformData],
+  );
+
+  if (!waveformData.length) return null;
+
+  return (
+    <div
+      className='str-chat__wave-progress-bar__track'
+      data-testid='wave-progress-bar-track'
+      onClick={seek}
+      onMouseDown={handleMouseDown}
+      onMouseLeave={handleMouseUp}
+      onMouseMove={handleMouseMove}
+      onMouseUp={handleMouseUp}
+      role='progressbar'
+    >
+      {resampledWaveformData.map((amplitude, i) => (
+        <div
+          className={clsx('str-chat__wave-progress-bar__amplitude-bar', {
+            ['str-chat__wave-progress-bar__amplitude-bar--active']:
+              progress > (i / resampledWaveformData.length) * 100,
+          })}
+          data-testid='amplitude-bar'
+          key={`amplitude-${i}`}
+          style={
+            {
+              '--str-cha__wave-progress-bar__amplitude-bar-height': amplitude
+                ? amplitude * 100 + '%'
+                : '0%',
+            } as React.CSSProperties
+          }
+        />
+      ))}
+      <div
+        className='str-chat__wave-progress-bar__progress-indicator'
+        data-testid='wave-progress-bar-progress-indicator'
+        ref={setProgressIndicator}
+        style={{ left: `${progress < 0 ? 0 : progress > 100 ? 100 : progress}%` }}
+      />
+    </div>
+  );
+};
+
+/**
+ * The downSample function uses the Largest-Triangle-Three-Buckets (LTTB) algorithm.
+ * See the thesis Downsampling Time Series for Visual Representation by Sveinn Steinarsson for more (https://skemman.is/bitstream/1946/15343/3/SS_MSthesis.pdf)
+ * @param data
+ * @param targetOutputSize
+ */
+export function downSample(data: number[], targetOutputSize: number): number[] {
```

Comment:
Downsampling algorithm used does not use mean to reduce the data, but it rather selects a value that represents the biggest difference compared to the previous and next bucket.